sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubClient/EventHubClientLiveTests.cs

```diff
@@ -215,5 +216,188 @@ await using (var client = new EventHubClient(connectionString))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString, new EventHubClientOptions { TransportType = TransportType.AmqpWebSockets }))
+                await using (var producer = client.CreateProducer())
+                {
+                    var events = new[] { new EventData(Encoding.UTF8.GetBytes("Awesome content")) };
+                    Assert.That(async () => await producer.SendAsync(events), Throws.Nothing);
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubClient" /> is able to
+        ///   connect to the Event Hubs service.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ClientCanReceiveWithAmqpWebSocketsConnection()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString, new EventHubClientOptions { TransportType = TransportType.AmqpWebSockets }))
+                {
+                    var partition = (await client.GetPartitionIdsAsync()).First();
+
+                    await using (var consumer = client.CreateConsumer(partition, EventPosition.Latest))
+                    {
+                        Assert.That(async () => await consumer.ReceiveAsync(1, TimeSpan.Zero), Throws.Nothing);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubClient" /> is able to
+        ///   connect to the Event Hubs service.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task ClientCannotRetrieveMetadataWhenClosed(bool sync)
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    var partition = (await client.GetPartitionIdsAsync()).First();
+
+                    Assert.That(async () => await client.GetPropertiesAsync(), Throws.Nothing);
+                    Assert.That(async () => await client.GetPartitionPropertiesAsync(partition), Throws.Nothing);
+
+                    if (sync)
+                    {
+                        client.Close();
+                    }
+                    else
+                    {
+                        await client.CloseAsync();
+                    }
+
+                    Assert.ThrowsAsync<OperationCanceledException>(async () => await client.GetPartitionIdsAsync());
```

Comment:
Can we please keep to the constraint form for these, so that we're consistent throughout the code?

```csharp
Assert.That(async () => await client.GetPartitionIdsAsync(), Throws.InstanceOf<OperationCancelledException>());
```

Reply:
Same for the rest, if you would be so kind.

Reply:
Done!